File: src/main/java/br/com/dusty/dkits/listener/mechanics/ItemSpawnListener.java
package br.com.dusty.dkits.listener.mechanics;

import br.com.dusty.dkits.util.TaskUtils;
import org.bukkit.entity.Item;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.ItemSpawnEvent;

public class ItemSpawnListener implements Listener {
	
	@EventHandler
	public void onItemSpawn(ItemSpawnEvent event) {
		Item item = event.getEntity();
		
		if(item.isValid())
			TaskUtils.sync(item::remove, 300);
	}
}

File: src/main/kotlin/br/com/dusty/dkits/listener/mechanics/ItemSpawnListener.kt
package br.com.dusty.dkits.listener.mechanics

import br.com.dusty.dkits.util.Tasks
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.ItemSpawnEvent

object ItemSpawnListener: Listener {

	@EventHandler
	fun onItemSpawn(event: ItemSpawnEvent) {
		val item = event.entity

		Tasks.sync(Runnable { item.remove() }, 100)
	}
}
